src/waypoint.jsx

```diff
@@ -202,23 +253,24 @@ export default class Waypoint extends React.Component {
       debugLog('scrollableAncestor height', contextHeight);
       debugLog('scrollableAncestor scrollTop', contextScrollTop);
     }
-    const thresholdPx = contextHeight * this.props.threshold;
+    const topOffsetPx = this._computeTopOffsetPx(contextHeight);
+    const bottomOffsetPx = this._computeBottomOffsetPx(contextHeight);
     const contextBottom = contextScrollTop + contextHeight;
 
     if (contextHeight === 0) {
       return Waypoint.invisible;
     }
 
-    if (contextScrollTop <= waypointTop + thresholdPx &&
-        waypointTop - thresholdPx <= contextBottom) {
+    if (contextScrollTop <= waypointTop - topOffsetPx &&
```

Comment:
I'll likely need to adjust the relational operators here. Getting unexpected results when just flipping the `+` to `-`.